### PR TITLE
Add openState in abstract importer

### PR DESCRIPTION
**Pull request by costashatz**
As discussed in gitter, this PR adds `openState` capability to the `AbstractImporter`. This can be used to directly set the state of the importer (i.e., to load from memory an already loaded file etc.).

I didn't try to add any doc in the `openState` function, as @mosra most probably knows better what to write to be consistent..

### CODE REVIEW

File: src/Magnum/Trade/AbstractImporter.h [hunk 1]
```diff
@@ -496,6 +496,8 @@ class MAGNUM_EXPORT AbstractImporter: public PluginManager::AbstractManagingPlug
          */
         const void* importerState() const;
 
+        void openState(const void* state);
```
Comment (Squareys) [inline src/Magnum/Trade/AbstractImporter.h]: There needs to be a piece of documentation here to explain what `openState` does 😉

Comment (mosra) [reply src/Magnum/Trade/AbstractImporter.h]: Similarly to the `*.cpp` file I would put this function right next to the `openData()`/`openFile()` declaration. Then you could just copypaste the docs from `openData()` and change relevant portions (mention `Feature::OpenState` etc.). That's it. There's then a few cases where `openData()`/`doOpanData()` is mentioned above in the main class docs, would be great if you'd mention the new functions there as well, but I can do that later, too.

File: src/Magnum/Trade/AbstractImporter.h [hunk 2]
```diff
@@ -821,6 +823,9 @@ class MAGNUM_EXPORT AbstractImporter: public PluginManager::AbstractManagingPlug
 
         /** @brief Implementation for @ref importerState() */
         virtual const void* doImporterState() const;
+
+        /** @brief Implementation for @ref openState() */
+        virtual void doOpenState(const void* state);
```
Comment (mosra) [inline src/Magnum/Trade/AbstractImporter.h]: Third and last comment: I would *also* put this next to the `doOpenData()`/`doOpenFile()` functions.

**Comment by costashatz** [comment]
> Okay, I think this is perfect now :)

Ahh..I just saw. Maybe we could document the `filePath` argument, or not?

**Comment by mosra** [comment]
I'll do that together with the other docs updates :)

**Comment by mosra** [comment]
Cherry-picked as 4d7f955d55f0b0e3d087106a840c2ecf2547c90e. Thank you!

**Pull request closed without merging**